aspnetcore/tutorials/signalr-typescript-webpack.md

```diff
@@ -0,0 +1,245 @@
+
+1. Execute the following command in the project root to create a *package.json* file:
+
+    ```console
+    npm init -y
+    ```
+
+1. Add the highlighted property to the *package.json* file. It prevents package installation warnings in the next step.
+
+    [!code-json[package.json](signalr-typescript-webpack/sample/snippets/package1.json?highlight=4)]
+
+1. Install the required npm packages. Execute the following command from the project root:
+
+    ```console
+    npm install -D -E clean-webpack-plugin@0.1.19 css-loader@0.28.11 html-webpack-plugin@3.2.0 mini-css-extract-plugin@0.4.0 ts-loader@4.4.1 typescript@2.9.2 webpack@4.12.0 webpack-cli@3.0.6
+    ```
+
+    Exact versions of npm packages, which are known to work, are installed.
+
+1. Replace the `scripts` property of the *package.json* file with the following snippet:
+
+    ```json
+    "scripts": {
+      "build": "webpack --mode=development --watch",
+      "release": "webpack --mode=production",
+      "publish": "npm run release && dotnet publish -c Release"
+    },
+    ```
+
+    Some explanation of the scripts:
+
+    * `build`: Bundles your client-side resources in development mode and watches for file changes. The file watcher causes the bundle to regenerate each time a project file changes. The `mode` option disables production optimizations, such as tree shaking and minification. Only use this script in development.
+    * `release`: Bundles your client-side resources in production mode.
+    * `publish`: Runs the `release` script to bundle the client-side resources in production mode. It calls the .NET Core CLI's [publish](/dotnet/core/tools/dotnet-publish) command to publish the app.
+
+1. Create a file named *webpack.config.js*, in the project root, with the following content. Its purpose is to configure the Webpack compilation.
+
+    [!code-javascript[webpack.config.js](signalr-typescript-webpack/sample/webpack.config.js)]
+
+    Some configuration details to note:
+
+    * The `output` property overrides the default value of *dist*. The bundle is instead emitted in the *wwwroot* directory.
+    * The `resolve.extensions` array includes *.js* to import the SignalR client JavaScript.
+
+1. Create a new *src* directory in the project root. Its purpose is to store the project's client-side assets.
+
+1. Create a file named *index.html*, in the *src* directory, with the following content. It defines the homepage's HTML template.
+
+    [!code-html[index.html](signalr-typescript-webpack/sample/src/index.html)]
+
+1. Create a file named *main.css*, in a new *src/css* directory, with the following content. It includes CSS classes for the app.
+
+    [!code-css[main.css](signalr-typescript-webpack/sample/src/css/main.css)]
+
+1. Create a file named *tsconfig.json*, in the *src* directory, with the following content. It configures the TypeScript compiler to produce [ECMAScript](https://wikipedia.org/wiki/ECMAScript) 5-compatible JavaScript.
```

Comment:
I prefer the following which is verbose and logically correct.

Create *src\tsconfig.json* with the following content:

`[code snippet import]`

The preceding code configures  the TypeScript compiler to produce [ECMAScript](https://wikipedia.org/wiki/ECMAScript) 5-compatible JavaScript.

Reply:
That's the format I'd like to follow